ftfy/fixes.py

```diff
@@ -82,42 +72,71 @@ def fix_text_encoding(text):
     0x81 that have no mapping in Windows. This is a string that Python's
     standard `.encode` and `.decode` methods cannot correct.
 
-        >>> print(fix_text_encoding('This text is sad .â\x81”.'))
+        >>> print(fix_encoding('This text is sad .â\x81”.'))
         This text is sad .⁔.
 
     However, it has safeguards against fixing sequences of letters and
-    punctuation that can occur in valid text:
+    punctuation that can occur in valid text. In the following example,
+    the last three characters are not replaced with a Korean character,
+    even though they could be.
 
-        >>> print(fix_text_encoding('not such a fan of Charlotte Brontë…”'))
+        >>> print(fix_encoding('not such a fan of Charlotte Brontë…”'))
         not such a fan of Charlotte Brontë…”
 
+    This function can now recover some complex manglings of text, such as when
+    UTF-8 mojibake has been normalized in a way that replaces U+A0 with a
+    space:
+
+        >>> print(fix_encoding('The more you know ðŸŒ '))
+        The more you know 🌠
+
     Cases of genuine ambiguity can sometimes be addressed by finding other
     characters that are not double-encoded, and expecting the encoding to
     be consistent:
 
-        >>> print(fix_text_encoding('AHÅ™, the new sofa from IKEA®'))
+        >>> print(fix_encoding('AHÅ™, the new sofa from IKEA®'))
         AHÅ™, the new sofa from IKEA®
 
     Finally, we handle the case where the text is in a single-byte encoding
     that was intended as Windows-1252 all along but read as Latin-1:
 
-        >>> print(fix_text_encoding('This text was never UTF-8 at all\x85'))
+        >>> print(fix_encoding('This text was never UTF-8 at all\x85'))
         This text was never UTF-8 at all…
 
     The best version of the text is found using
     :func:`ftfy.badness.text_cost`.
     """
-    text, _plan = fix_encoding_and_explain(text)
+    text, _ = fix_encoding_and_explain(text)
     return text
 
 
+def fix_text_encoding(text):
+    """
+    A deprecated name for :func:`ftfy.fixes.fix_encoding`.
+    """
+    warnings.warn('fix_text_encoding is now known as fix_encoding',
+                  DeprecationWarning)
+    return fix_encoding(text)
+
+
+# When we support discovering mojibake in more encodings, we run the risk
+# of more false positives. We can mitigate false positives by assigning an
+# additional cost to using encodings that are rarer than Windows-1252, so
+# that these encodings will only be used if they fix multiple problems.
+ENCODING_COSTS = {
+    'macroman': 2,
+    'cp437': 3,
+    'sloppy-windows-1251': 5
+}
+
+
 def fix_encoding_and_explain(text):
     """
     Re-decodes text that has been decoded incorrectly, and also return a
     "plan" indicating all the steps required to fix it.
 
     To fix similar text in the same way, without having to detect anything,
-    you can use the ``apply_plan`` function.
+    you can use the :func:`ftfy.fixes.apply_plan` function.
```

Comment:
I think what this is trying to say is that if you have a bunch of different texts that you know are all broken in the same way, you can run this function on one of them to get the "plan", and then apply that plan to all the other texts without having to re-derive it?  I think this docstring sentence could be clearer though, about what it means by "similar text".